hera_cal/io.py

```diff
@@ -430,20 +441,39 @@ def get_blt_slices(uvo, tried_to_reorder=False):
     Returns:
         blt_slices: dictionary mapping anntenna pair tuples to baseline-time slice objects
     '''
-    blt_slices = {}
-    for ant1, ant2 in uvo.get_antpairs():
-        indices = uvo.antpair2ind(ant1, ant2)
-        if len(indices) == 1:  # only one blt matches
-            blt_slices[(ant1, ant2)] = slice(indices[0], indices[0] + 1, uvo.Nblts)
-        elif not (len(set(np.ediff1d(indices))) == 1):  # checks if the consecutive differences are all the same
-            if not tried_to_reorder:
-                uvo.reorder_blts(order='time')
-                return get_blt_slices(uvo, tried_to_reorder=True)
+    if assume_bl_first and assume_time_first:
+        raise ValueError("Cannot assume both bl first and time first ordering!")
+    if assume_bl_first:
+        if not uvo.Nblts != uvo.Nbls * uvo.Ntimes:
+            raise ValueError("Assumption of bl first ordering cannot be correct if Nblts != Nbls * Ntimes")
+        for i in range(uvo.Nbls):
+            antp = (uvo.ant_1_array[i], uvo.ant_2_array[i])
+            blt_slices[antp] = slice(i, uvo.Nblts, uvo.Nbls)
+        assert uvo.Nbls == len(blt_slices)
+
+    elif assume_time_first:
+        if not uvo.Nblts != uvo.Nbls * uvo.Ntimes:
+            raise ValueError("Assumption of time first ordering cannot be correct if Nblts != Nbls * Ntimes")
+        for i in range(uvo.Nbls):
+            antp = (uvo.ant_1_array[i*uvo.Ntimes], uvo.ant_2_array[i*uvo.Ntimes])
+            blt_slices[antp] = slice(i, uvo.Nbls + i, 1)
+        assert uvo.Nbls == len(blt_slices)
```

Comment:
Not wild about none of this being covered by tests. Do we need it?

Reply:
This speeds up getting the blt slices quite significantly, which is very useful when reading in a lot of files. In saying that, I've been able to clean up the code a little bit here because pyuvdata does some of the heavy lifting now (it'll do even more of it in the future). I'll make sure this gets covered by tests in the meantime.